### ISSUE TITLE
Azure: ResourceGroupNotFound on v1.11.0

### ISSUE THREAD

**Issue by jcrowthe** [kubernetes/kubernetes#65651]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug
@kubernetes/sig-azure-bugs

**What happened**:
The Azure cloud provider code used by `kube-controller-manager` appears to fail to reach out to Azure on all requests, starting in v1.11.0. Error: 
> Status=404 Code=\"ResourceGroupNotFound\" Message=\"Resource group 'kube59' could not be found.\""

Log snippet:
```
I0629 19:53:06.364743       1 azure_wrap.go:263] Route table "routes-kube59" not found with message: "network.RouteTablesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code=\"ResourceGroupNotFound\" Message=\"Resource group 'kube59' could not be found.\""
I0629 19:53:06.364785       1 azure_routes.go:84] createRouteTableIfNotExists: creating routetable. routeTableName="routes-kube59"
I0629 19:53:06.392953       1 azure_wrap.go:263] Route table "routes-kube59" not found with message: "network.RouteTablesClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code=\"ResourceGroupNotFound\" Message=\"Resource group 'kube59' could not be found.\""
I0629 19:53:06.393065       1 azure_routes.go:84] createRouteTableIfNotExists: creating routetable. routeTableName="routes-kube59"
```

Cloud provider config:
```
---
cloud: AzurePublicCloud
subscriptionId: 84b...
resourceGroup: kube59
location: eastus2
subnetName: subnet-agentkube59
securityGroupName: nsg-edge-kube59
vnetName: vnet-kube59
vnetResourceGroup: kube59
vmType: vmss
routeTableName: routes-kube59
primaryScaleSetName: vmss-agent-worker-kube59
cloudProviderBackoff: false
cloudProviderBackoffRetries: 0
cloudProviderBackoffExponent: 0
cloudProviderBackoffDuration: 0
cloudProviderBackoffJitter: 0
cloudProviderRatelimit: false
cloudProviderRateLimitQPS: 0
cloudProviderRateLimitBucket: 0
useManagedIdentityExtension: true
useInstanceMetadata: true
```

**What you expected to happen**:
kube-controller-manager would properly talk to Azure ARM API to create/configure resources such as LB, routes, NSG rules, etc.

**How to reproduce it (as minimally and precisely as possible)**:
Manually launch a k8s cluster in Azure, with MSI enabled. (not AKS or ACS since they do not yet support v1.11.0)

**Anything else we need to know?**:
I do not believe this is MSI related. I can confirm through using az cli that the token is properly created:
```
$ az role assignment list-changelogs
...
{
    "action": "Granted",
    "caller": "<email>",
    "principalId": "d0...",
    "principalName": "5947b...",
    "roleDefinitionId": "b24988...",
    "roleName": "Contributor",
    "scope": "/subscriptions/84b.../resourcegroups/kube59",
    "scopeName": "kube59",
    "scopeType": "Resource group",
    "timestamp": "2018-06-29T18:25:26.767813+00:00"
  }
]
```

Interesting find: in proving out 1.11, this issue initially appeared in the kube-apiserver, with the same error, `ResourceGroupNotFound`, which caused it to fail at start time. However, once the cloud config was updated to reflect `useInstanceMetadata: true`, apiserver then went healthy without further issues. In investigating the next discovered issue (kube-controller-manager not deploying cloud resources), neither setting `useInstanceMetadata` to true or false had an effect.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11.0
- Cloud provider or hardware configuration: Azure
- OS (e.g. from /etc/os-release): CoreOS 1745.7.0
- Kernel (e.g. `uname -a`): `Linux vmss-master-kube59000001 4.14.48-coreos-r2 #1 SMP Thu Jun 14 08:23:03 UTC 2018 x86_64 Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz GenuineIntel GNU/Linux`
- Install tools:
- Others:

**Comment by feiskyer** [comment]
@jcrowthe Are there any other errors besides ResourceGroupNotFound one?

**Comment by jcrowthe** [comment]
Not that I could see. I checked the logs for kube-apiserver -controller-manager -scheduler and -proxy, but only saw the few lines listed above, or ones very similar to it.

**Comment by jcrowthe** [comment]
Issue appears to have resolved itself, either through deploying v1.11.0 or Azure changed something.
